### PR TITLE
add to muparser docs

### PR DESCRIPTION
**Pull request by bobmyhill**
This PR improves the documentation of the Function expression, adding a link to the muparser documentation and a python binding for muparser.

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### CODE REVIEW

File: doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
 
+## Help with MuParser syntax
+If you would like to check that muparser is parsing your functions correctly,
+and to plot the values, you may consider running your expressions through
+`pymuparser` (<https://github.com/bobmyhill/pymuparser>),
+which can be installed on your machine using pip
+(`python -m pip install pymuparser`). If you need to use functions in the
+extended library provided by deal.II, Python's `eval` function may be useful,
+(<https://docs.python.org/3/library/functions.html#eval>)
+but be aware that python syntax may not be the same as that of deal.II.
```
Comment (tjhei) [inline doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md:38]:
```suggestion
but be aware that Python syntax may not be the same as that of deal.II.
```

File: doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md [hunk 2]
```diff
@@ -51,16 +62,14 @@ as `((1<x && x<4) ? ((2<y && y<3) ? 2 : 1) : (0))`.
 
 An example for how to translate nested if-else statements into the
 lazy-expression syntax is given in the cookbook example found in
-{ref}`sec:cookbooks:muparser_temperature_example`. This cookbook includes a python script that defines
+{ref}`sec:cookbooks:muparser_temperature_example`.
+This cookbook includes a python script that defines
```
Comment (tjhei) [inline doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md:66]:
```suggestion
This cookbook includes a Python script that defines
```

**Review by tjhei — APPROVED** [review]

**Comment by tjhei** [comment]
Very nice. Can you fix the capitalization though?

File: doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md [hunk 3]
```diff
@@ -1,66 +1,71 @@
...
+including `int()`, `ceil()`, `floor()`, `cot()`, `csc()`, `sec()`,
+`pow()`, `erfc()`, `rand()`, and `rand_seed()`. For more detailed information, see
 <http://www.dealii.org/developer/doxygen/deal.II/classFunctionParser.html#details>.
 
-A common need for function expression is an if-else-statement, for example
-"if $1<x<4$ then output 1, else output 0." The muparser uses
+## Help with MuParser syntax
+If you would like to check that muparser is parsing your functions correctly,
+and to plot the values, you may consider running your expressions through
+`pymuparser` (<https://github.com/bobmyhill/pymuparser>),
```
Comment (bangerth) [inline doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md:33]: That's a very nice project! Would you want to also write a pull request for deal.II's documentation with similar information?

File: doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md [hunk 4]
```diff
@@ -1,66 +1,71 @@
...
+for example "if $1<x<4$ then output 1, else output 0". MuParser uses
 lazy-expression syntax `(if-condition ? true-expression : false-expression)`
 for if-else statements. This lazy-expression only evaluates the expression
 that meets the if-condition, rather than evaluating both expressions, which
 can be useful if one of the expressions is not defined (e.g., has a divide by
 zero) when the if-condition is not met. Note it is also possible to use the
 syntax `if(condition, true-expression, false-expression)`, but in this case
 both expressions are always evaluated. This is inefficient, but in addition
-may abort the program with a floating point exception if the expression that
+may yield avoidable floating point exceptions if the expression that
```
Comment (bangerth) [inline doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md:50]: I had no idea about the difference!

**Review by bangerth — APPROVED** [review]

File: doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md (hunk 3 above)
Comment (bobmyhill) [reply doc/sphinx/user/run-aspect/parameters-overview/muparser-format.md:33]: Sure, I can do that. I was also wondering about extending the module with the additional deal.II functions. It wouldn't be hard to add some nesting logic, but this is probably not a top priority for the next couple of days.

**Pull request merged**